### PR TITLE
Quick pass at caching the insights page

### PR DESCRIPTION
**Pull request by adelevie**
An attempt at page caching `/insights` using https://github.com/rails/actionpack-page_caching.

To test this locally, set `config.action_controller.perform_caching = true` in `config/environments/development.rb`, run the server and visit `/insights`. Check to see that `public/cache/insights.html` exists. To force cache expiration, open a rails console and run `ClearCache.new.perform` and check that `public/cache/insights.html` no longer exists.

### CODE REVIEW

File: app/services/clear_cache.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class ClearCache
+  def perform
+    ApplicationController.expire_page(app.insights_path)
```
Comment (jessieay) [inline app/services/clear_cache.rb]: cool! have we confirmed that this works?

File: config/clock.rb [hunk 2]
```diff
@@ -18,4 +18,9 @@ module Clockwork
     puts "Checking for approved auctions"
     CheckApproval.new.delay.perform
   end
+
+  every(1.day, "insights_cache.clear") do
+    puts "Clearing the page cache for the Insights page"
+    ClearCache.new.delay.perform
```
Comment (jessieay) [inline config/clock.rb]: niiiice

**Comment by jessieay** [comment]
looks good! code climate looks like an easy fix

**Comment by adelevie** [comment]
@jessieay I confirmed this locally from a manual test, but does this seem like something that needs an automatic test? E.g. I have not yet simulated the clock.rb stuff, just ran `ClearCache#perform`.

**Comment by jessieay** [comment]
@adelevie I do think one thing that could be cool here would be to add an integration test that looks for a stat, changes the auction data, clears cache, and looks for new stat. thoughts?

File: spec/services/clear_cache_spec.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# require 'rails_helper'
```
Comment (jessieay) [inline spec/services/clear_cache_spec.rb]: delete?

File: spec/factories/auctions.rb [hunk 4]
```diff
@@ -116,6 +116,7 @@
     end
 
     trait :accepted do
+      ended_at { Time.now - 1.days }
```
Comment (jessieay) [inline spec/factories/auctions.rb]: what is this related to?

File: features/support/env.rb [hunk 5]
```diff
@@ -23,6 +23,16 @@
   Delayed::Worker.delay_jobs = false
 end
 
+Before '@enable_caching' do
+  ActionController::Base.perform_caching = true
```
Comment (jessieay) [inline features/support/env.rb]: interesting -- is this not usually enabled in test env?

File: features/step_definitions/auction_create_steps.rb [hunk 6]
```diff
@@ -174,11 +174,11 @@
...
-    :published,
-    result: :accepted,
-    accepted_at: nil,
-  )
+  @auction = FactoryGirl.create(:auction, :accepted, :with_bidders)
+end
+
+When(/^there is another accepted auction$/) do
+  # not creating another @auction-like instance variable
+  # just creating the record without saving to a variable
```
Comment (jessieay) [inline features/step_definitions/auction_create_steps.rb]: I think think these comments are needed

File: spec/services/clear_cache_spec.rb (hunk 3 above)
Comment (adelevie) [inline spec/services/clear_cache_spec.rb]: sorry, still sketching ... I'm going to add a simple unit test, too.

File: spec/factories/auctions.rb (hunk 4 above)
Comment (adelevie) [inline spec/factories/auctions.rb]: For seed data to be displayed in /insights, I thought the auction needs to have ended to be counted. But `result: :accepted` might be sufficient here.

File: features/support/env.rb (hunk 5 above)
Comment (adelevie) [inline features/support/env.rb]: as of now, `config/environments/test.rb` does not have `perform_caching` anywhere. Seems safer to explicitly test only where needed? Or maybe have test match prod and apply caching globally in the test env?

Comment (jessieay) [inline features/support/env.rb]: gotcha -- this seems oK for now!

File: spec/factories/auctions.rb (hunk 4 above)
Comment (adelevie) [inline spec/factories/auctions.rb]: @jessieay I think also, as a rule, wouldn't an accepted auction always have an ended_at timestamp? And if so, shouldn't the factory reflect this?

**Comment by adelevie** [comment]
Not as great as I'd like it to be, but it's ready imo, and we should at least validate that this will work on staging before much more work is done.

File: spec/services/clear_cache_spec.rb (hunk 3 above)
Comment (jessieay) [inline spec/services/clear_cache_spec.rb]: did you want to write a unit test or not?

**Comment by jessieay** [comment]
looks good @adelevie  -- can we just delete that empty spec file?

**Comment by adelevie** [comment]
good catch--deleted.

**Comment by jessieay** [comment]
merging!! 🌈

**Comment by adelevie** [comment]
yay!
